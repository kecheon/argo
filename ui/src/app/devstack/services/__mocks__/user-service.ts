
const userInfo = {
    loggedIn: true,
    username: 'devstack',
    iss: 'some_iss',
    groups: ['devstack1', 'devstack2']
};

export class UserService {
    public login = jest.fn(() => Promise.resolve({data: userInfo, headers: { 'x-subject-token': 'unscoped_token_string'}}));
    public register = jest.fn(() => Promise.resolve({data: {token: 'token_string'}}));
    public logout = jest.fn(() => Promise.resolve({data: {message: 'logout ok'}}));
}
