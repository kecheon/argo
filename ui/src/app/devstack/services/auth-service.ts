import {AuthSettings} from '../../shared/models';
import requests from '../../shared/services/requests';

export class AuthService {
    public settings(): Promise<AuthSettings> {
        return requests.get('/settings').then(res => res.body as AuthSettings);
    }
}
