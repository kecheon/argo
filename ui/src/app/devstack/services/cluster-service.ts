import axios from 'axios';
const accessToken = localStorage.getItem('accessToken');
import {endpoint} from '../../devstack/classes/constants';
const headers = {
   Authorization: `Bearer ${accessToken}` 
}
export class ClusterService {
    public async get(): Promise<any> {
        try {
            const response = await axios.get(`${endpoint}/cluster`, { headers });
            return response.data;
        } catch(err) {
            return [];
        }
    }
    public async getDetail(id: string): Promise<any> {
        try {
            const response = await axios.get(`${endpoint}/cluster/${id}`, { headers });
            return response.data;
        } catch(err) {
            return [];
        }
    }
    public async create(data: any): Promise<any> {
        try {
            const response = await axios.post(`${endpoint}/cluster`, data, {headers});
            return response.data;
        } catch(err) {
            return { status: 'failure' };
        }
    }
}
