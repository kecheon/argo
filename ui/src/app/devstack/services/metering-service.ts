import axios from 'axios';
import {endpoint} from '../classes/constants';
const accessToken = localStorage.getItem('accessToken');
const headers = {
   Authorization: `Bearer ${accessToken}` 
}
export class MeteringService {
    public async get(minStartedAt: any, maxStartedAt: any): Promise<any> {
        // const start = startDate.toString();
        // const end = endDate.toString();
        const params = { minStartedAt, maxStartedAt };
        const response = await axios.get(`${endpoint}/argo/metering`, { params, headers });
        console.log(response);
        return response.data;
    }
    public async getByNamespace(namespace: string, minStartedAt: any, maxStartedAt: any): Promise<any> {
        const params = { minStartedAt, maxStartedAt };
        const response = await axios.get(`${endpoint}/argo/metering/${namespace}`, { params, headers });
        return response.data;
    }
    public async getWorkflows(): Promise<any> {
        const response = await axios.get(`${endpoint}/argo/overview-workflows`, { headers });
        return response.data;
    }
}
