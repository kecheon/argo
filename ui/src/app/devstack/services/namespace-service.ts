import axios from 'axios';
import namespacesList from './__mocks__/namespaces';
// import { UsersList } from '../users/components/users-list/users-list';
import {endpoint} from '../../devstack/classes/constants';
// const keystoneEndPoint = 'http://183.111.177.141/identity/v3';
import {NamespaceForm} from '../users/components/namespaces/components/create-namespace/create-namespace';
const accessToken = localStorage.getItem('accessToken');
const headers = {
   Authorization: `Bearer ${accessToken}` 
}
export class NamespaceService {
    public async get(): Promise<any> {
        // get projects list if wf-admin

        const response = await axios.get(`${endpoint}/namespace`, { headers });
        return response.data;
    }
    public async delete(id: string): Promise<any> {
        const response = await axios.delete(`${endpoint}/namespace/${id}`, { headers });
        return { status: 'success', message: 'namespace deleted' }
    }
    public async create(data: NamespaceForm): Promise<any> {
        const response = await axios.post(`${endpoint}/namespace`, data, { headers });
        return { status: 'success', message: 'namespace created' }
    }
    public async update(data: NamespaceForm): Promise<any> {
        const response = await axios.put(`${endpoint}/namespace`, data, { headers });
        return { status: 'success', message: 'namespace created' }
    }
    public async getProfile(id: string): Promise<any> {
        const response = await axios.get(`${endpoint}/namespace/${id}`, { headers });
        // return {
        //     'is_wf': true,
        //     'wf': {
        //         'k8s_ns': 'dl-inferencing',
        //         'quota_cpu': 10,
        //         'quota_ram': 16
        //     },
        //     'id': '978a647141a04db2a28c19eba924f6c0',
        //     'name': 'dl-inferencing',
        //     'domain_id': 'default',
        //     'description': 'inference',
        //     'enabled': true,
        //     'parent_id': 'default',
        //     'is_domain': false,
        //     'tags': [],
        //     'options': {},
        //     'links': {
        //         'self': 'http://183.111.177.141/identity/v3/projects/978a647141a04db2a28c19eba924f6c0'
        //     }
        // };
        return response.data;
    }
    public async getMembers(id: string): Promise<any> {
        const response = await axios.get(`${endpoint}/namespace/${id}/member`, { headers });
        console.log(response);
        return response.data;
    }

    public async updateMember(id: string, data: any): Promise<any> {
        const response = await axios.patch(`${endpoint}/namespace/${id}/member`, data, { headers });
        console.log(response);
        return response;
    }

    public async updateNamespace(namespace: string, namespaceId: string, data: any): Promise<any> {
        const response = await axios.patch(`${endpoint}/namespace/${namespaceId}`, data, { headers });
        console.log(response);
        return response;  
    }
}
