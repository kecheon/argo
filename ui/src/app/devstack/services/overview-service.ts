import axios from 'axios';
import {endpoint} from '../../devstack/classes/constants';
const accessToken = localStorage.getItem('accessToken');
const headers = {
   Authorization: `Bearer ${accessToken}` 
}
export class OverviewService {
    public async get(): Promise<any> {
        const response = await axios.get(`${endpoint}/argo/overview-data`, { headers });
        return response.data;
    }
    public async getByNamespace(namespace: string): Promise<any> {
        const response = await axios.get(`${endpoint}/argo/overview-data/${namespace}`, { headers });
        return response.data;
    }
    public async getWorkflows(): Promise<any> {
        const response = await axios.get(`${endpoint}/argo/overview-workflows`, { headers });
        return response.data;
    }
    public async getWorkflowsByNamespace(namespace: string): Promise<any> {
        const response = await axios.get(`${endpoint}/argo/overview-workflows/${namespace}`, { headers });
        return response.data;
    }
}
