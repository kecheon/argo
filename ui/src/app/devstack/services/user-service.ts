import axios from 'axios';
import {UserInfo} from '../../shared/models';
const endPoint = 'http://183.111.177.141/identity/v3/';

export class UserService {
    public async login(username: string, password: string): Promise<any> {
        const reqBody = {
            auth: {
                identity: {
                    methods: ['password'],
                    password: {
                        user: {
                            name: username,
                            domain: {id: 'default'},
                            password
                        }
                    }
                }
            }
        };
        const response = await axios.post(`${endPoint}/api/v1/auth`, reqBody);
        return response.data;
    }

    public async logout(): Promise<boolean> {
        const response = await axios.delete('/api/v1/auth');
        return response.data;
    }

    public async get(): Promise<UserInfo> {
        const response = await axios.get('/api/v1/auth/userinfo', {params: {token: 'token_string'}});
        return response.data as UserInfo;
    }
    public async register(userProfile: object): Promise<UserInfo> {
        const response = await axios.post('/api/v1/auth/register', userProfile);
        return response.data as UserInfo;
    }
}
