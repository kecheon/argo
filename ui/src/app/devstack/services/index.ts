import {AuthService} from './auth-service';
import {UserService} from './user-service';

export interface Services {
    users: UserService;
    authService: AuthService;
}

export const services: Services = {
    users: new UserService(),
    authService: new AuthService()
};
