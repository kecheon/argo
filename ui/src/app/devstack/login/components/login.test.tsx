/*
 // current testing env is node on jest.config so that need to set this test's env to jsdom
 * @jest-environment jsdom
*/
import axios from 'axios';

import '@testing-library/jest-dom/extend-expect';
// import {render} from '@testing-library/react';
import {mount} from 'enzyme';
import * as React from 'react';
// import {BrowserRouter as Router} from 'react-router-dom';
import Login from './login';
import {UserService} from '../../services/user-service';

const nextTick = async () => {
    return new Promise(resolve => {
        setTimeout(resolve, 0);
    })
}

describe('Login component', () => {
    test('renders the component', () => {
        const component = mount(<Login />);
        expect(component).toMatchSnapshot();
    });
    test('enter login credentials and redirect to workflows url', async () => {
        const wrapper = mount(<Login />);
        const input1 = wrapper.find('input[name="username"]');
        input1.simulate('change', {
            target: {
                value: 'admin@devstack.co.kr'
            }
        });
        expect(wrapper.find('input[name="username"]').prop('value')).toEqual('admin@devstack.co.kr');
        const input2 = wrapper.find('input[name="password"]');
        input2.simulate('change', {
            target: {
                value: 'devstack'
            }
        });
        expect(wrapper.find('input[name="password"]').prop('value')).toEqual('devstack');

    });
    xtest('mock userService login', async () => {
        /* 
            You can't test handleSubmit function directly. The nature of the JavaScript language means that you can never access things hidden inside a closure.
            In this case, you can test it by shallow rendering Login, and asserting on the non-button children of the div it renders.
        */
        const service = new UserService();
        const mock = jest.spyOn(service, 'login');
        const wrapper = mount(<Login />);
        const submitBtn = wrapper.find('button[type="submit"]');
        expect(submitBtn.length === 1).toBeTruthy();
        submitBtn.simulate('click');
        await nextTick();
        expect(mock).toHaveBeenCalled();
    });
});


describe('Login api of Gateway', () => {
    const endpoint = 'http://localhost:3000';
    test('responses something precious', async () => {
        const credential = {
            username: 'admin',
            domainId: 'default',
            password: 'devstack'
        }
        const res = await axios.post(`${endpoint}/account/login`, credential);
        expect(res.status).toEqual(200);
        console.log(res);

    })
})