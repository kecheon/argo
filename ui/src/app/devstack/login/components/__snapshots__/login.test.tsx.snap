// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the component 1`] = `ReactWrapper {}`;
